spec/shared/controllers/hiding_columns_tenant.rb

```diff
@@ -0,0 +1,39 @@
+shared_examples 'hiding tenant columns' do
+
+  before do
+    allow(controller).to receive(:render)
+    login_as user
+  end
+
+  subject do
+    expect(controller).to receive(:get_db_view).and_return(report)
+    controller.send(:report_data)
+    controller.send(:view_to_hash, assigns(:view))
+  end
+
+  context "when not admin" do
+    let(:user) { FactoryBot.create(:user) }
+    it "renders show_list and does not include hidden column(hidden by display method)" do
+      headers = subject[:head].map { |x| x[:text] }.compact
+      expect(headers).to match_array(result_header)
+      expect(headers).not_to include('Tenant')
+
+      first_row_values = subject[:rows][0][:cells].map { |x| x[:text] }.compact
+      expect(first_row_values).to match_array(result_array)
+      expect(first_row_values).not_to include(not_in_array)
+    end
+  end
+
+  context "when admin" do
+    let(:user) { FactoryBot.create(:user_admin) }
+
+    it "renders show_list and includes all columns" do
+      controller.send(:report_data)
+      headers = subject[:head].map { |x| x[:text] }.compact
+      expect(headers).to match_array(result_header_admin)
+
+      first_row_values = subject[:rows][0][:cells].map { |x| x[:text] }.compact
+      expect(first_row_values).to match_array(result_array + [not_in_array])
```

Comment:
If you rename the `let!` variable to be the same in each test above, you can easily clean substitute `not_in_array` variable:
```
result_array + [record.tenant.name]
```